lib/Fhp/Dialog/Dialog.php

```diff
@@ -389,9 +387,9 @@ public function initDialog()
 			array(
 				$identification,
 				$prepare,
-				new HKTAN(HKTAN::VERSION, 5)
+				new HKTAN(HKTAN::VERSION, 5, null, $tanMediaName)
 			),
-			array(AbstractMessage::OPT_PINTAN_MECH => $this->supportedTanMechanisms)
+			array(AbstractMessage::OPT_PINTAN_MECH => array_keys($this->supportedTanMechanisms))
```

Comment:
`$this->supportedTanMechanisms` is a dictionary now with the key being the code as provided by the bank and the name as value. Formerly it was just the code, parsed in two different places.

Reply:
:heavy_check_mark: